smartmeter_datacollector/smartmeter/cosem.py

```diff
@@ -19,19 +37,59 @@ class RegisterCosem:
 
 
 class CosemConfig:
-    def __init__(self, id_obis: str, clock_obis: str, register_obis: List[RegisterCosem]) -> None:
-
-        self._id_obis = id_obis
-        self._clock_obis = clock_obis
+    def __init__(self, fallback_id: str, register_obis: List[RegisterCosem]) -> None:
+        self._id: Optional[str] = None
+        self._fallback_id = fallback_id
         self._register_obis = {r.obis: r for r in register_obis}
 
-    @property
-    def id_obis(self) -> str:
-        return self._id_obis
+    def retrieve_id(self, dlms_objects: Dict[str, Any]) -> str:
+        if self._id:
```

Comment:
If we use the `fallback_id` we will run through this logic for every received data packet. Should we avoid this or is it possible that we first don't receive any ID (and use the `fallback_id`) and later on we will get a real ID?

Reply:
Yes, but I inserted a countdown now. It's not that ugly after all, I think ;).